### ISSUE TITLE
Potentially spurrious

### ISSUE THREAD

**Issue by chloestefantsova** [dart-lang/sdk#56963]
Consider the following code:

```dart
extension on int? {
  int? operator+(int? other) {
    if (this != null && other != null) {
      return this! + other;
    } else {
      return null;
    }
  }
}

int test(int? x) {
  return ++x ?? 10;
}
```

The expected static type of `++x` is `int?`, given the return type of `operator+`. However, the Analyzer reports the following warning:

```
warning - qwerty.dart:12:17 - The left operand can't be null, so the right operand is never executed. Try removing the operator and the right operand. - dead_null_aware_expression
```

**Comment by adenes** [comment]
I hope you don't mind me chiming in. I did some debugging and it seems this is where the nullability from `int?` gets lost: https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/lib/src/dart/resolver/prefix_expression_resolver.dart#L231-L232

**Comment by bwilkerson** [comment]
> I hope you don't mind me chiming in.

No, we don't mind. We welcome all contributions. Thanks for digging into this and letting us know what you found!

**Comment by adenes** [comment]
https://dart-review.googlesource.com/c/sdk/+/392905

**Comment by eernstg** [comment]
@chloestefantsova, I commented on the CL (not that I could say much more than "I agree!" ;-).
